### PR TITLE
if not pruning, set preallocation of block/undo files to their max size

### PR DESCRIPTION
**Pull request by Greg-Griffith**

### CODE REVIEW

**Comment by sickpig** [comment]
code looks good to me. Could you explain to me the rationale for these changes?

**Comment by Greg-Griffith** [comment]
@sickpig assuming we are not pruning....
the max size we will let a blockfile grow to is 0x8000000 (128MB). Currently when we allocate space for block files we do it in 0x1000000 (16MB) chunks. This means that over the course of syncing for any given block file we will end up increasing the size 8 times before generating a new file. I think it makes more sense, since we will use almost the entire 128MB every time anyway, to just allocate the entire max size all at once. In general this should save some time at the cost of a small amount of additional disk space although i would think that if you are not pruning your node then disk space isn't an issue for you anyway. 

I believe the only time this actually "wastes space" is if the latest block file is already larger than 112MB and the next block that comes in is > 16MB in which case we would have allocated an extra 16 MB to the file that wont be used since all data for a block needs to be in the same file.

**Review by sickpig — APPROVED** [review]

File: src/main.h [hunk 1]
```diff
@@ -89,9 +89,11 @@ static const unsigned int DEFAULT_ORPHANPOOL_EXPIRY = 4;
 /** The maximum size of a blk?????.dat file (since 0.8) */
 static const unsigned int MAX_BLOCKFILE_SIZE = 0x8000000; // 128 MiB
 /** The pre-allocation chunk size for blk?????.dat files (since 0.8) */
-static const unsigned int BLOCKFILE_CHUNK_SIZE = 0x1000000; // 16 MiB
+static const unsigned int DEFAULT_BLOCKFILE_CHUNK_SIZE = 0x1000000; // 16 MiB
 /** The pre-allocation chunk size for rev?????.dat files (since 0.8) */
-static const unsigned int UNDOFILE_CHUNK_SIZE = 0x100000; // 1 MiB
+static const unsigned int DEFAULT_UNDOFILE_CHUNK_SIZE = 0x100000; // 1 MiB
+extern unsigned int blockfile_chunk_size;
+extern unsigned int undofile_chunk_size;
```
Comment (ptschip) [inline src/main.h]: can we move these params to blockstorage.h? I don't think they're used anywhere else anymore.

Comment (Greg-Griffith) [reply src/main.h]: yes, will do

**Comment by gandrewstone** [comment]
This will tend to lay the full file out sequentially on spinning disks which will have at least theoretical performance improvements when providing blocks since sequential read time is a lot faster than seek time.  But in practice it probably won't matter much until blocks are above 8MB.  At that point (without this PR), with 16MB "boundaries" in the file as laid out on disk, every other block starts requiring 2 seeks rather than just 1.  As block sizes approach 16MB every block needs 2 seeks.

**Pull request merged**